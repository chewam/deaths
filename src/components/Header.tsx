import Menu from "./Menu"
import Link from "next/link"
import { FormattedMessage as Trans } from "react-intl"

const Header = () => (
  <div className="header">
    <div className="wrapper">
      <h1>
        <Link href="/">
          <Trans id="Mortality in France" />
        </Link>
      </h1>
      <Menu />
    </div>
  </div>
)

export default Header
