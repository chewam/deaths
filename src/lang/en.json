{
  "overview": "Overview",
  "comparison": "Comparison",
  "distribution": "Distribution",
  "Mortality rate": "Mortality rate",
  "Deaths count": "Deaths count",
  "Total population": "Total population",
  "Mortality in France": "Mortality in France",
  "INSEE data": "INSEE data",
  "deaths": "deaths",
  "males": "males",
  "females": "females",
  "Age group": "Age group",
  "to": "to",
  "years old": "years old",
  "light mode": "light mode",
  "dark mode": "dark mode",
  "January": "January",
  "February": "February",
  "March": "March",
  "April": "April",
  "May": "May",
  "June": "June",
  "July": "July",
  "August": "August",
  "September": "September",
  "October": "October",
  "November": "November",
  "December": "Decembere"
}
