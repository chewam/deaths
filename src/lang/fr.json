{
  "overview": "Vue d'Ensemble",
  "comparison": "Comparaison",
  "distribution": "Répartition",
  "Mortality rate": "Taux de mortalité",
  "Deaths count": "Nombre de décès",
  "Total population": "Population totale",
  "Mortality in France": "Mortalité en France",
  "INSEE data": "Données INSEE",
  "deaths": "décès",
  "males": "hommes",
  "females": "femmes",
  "Age group": "Tranche d'âge",
  "to": "à",
  "years old": "ans",
  "light mode": "mode clair",
  "dark mode": "mode sombre",
  "January": "Janvier",
  "February": "Février",
  "March": "Mars",
  "April": "Avril",
  "May": "Mai",
  "June": "Juin",
  "July": "Juillet",
  "August": "Aout",
  "September": "Septembre",
  "October": "Octobre",
  "November": "Novembre",
  "December": "Decembre"
}
