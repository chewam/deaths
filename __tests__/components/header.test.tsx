/**
 * @jest-environment jsdom
 */
import fr from "@/lang/fr.json"
import en from "@/lang/en.json"
import Header from "@/components/Header"
import { IntlProvider } from "react-intl"
import { render } from "@testing-library/react"

const messages = { en, fr }

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      locale: "fr",
      defaultLocale: "en",
    }
  },
}))

/* eslint @typescript-eslint/no-var-requires: "off" */
const useRouter = jest.spyOn(require("next/router"), "useRouter")

describe("Header", () => {
  test("should create a Header in french", () => {
    const lang = "fr"

    useRouter.mockImplementation(() => ({
      route: "/",
    }))

    const { asFragment } = render(
      <IntlProvider locale={lang} messages={messages[lang]}>
        <Header />
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should create a Header in english", () => {
    const lang = "en"

    useRouter.mockImplementation(() => ({
      route: "/",
    }))

    const { asFragment } = render(
      <IntlProvider locale={lang} messages={messages[lang]}>
        <Header />
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
