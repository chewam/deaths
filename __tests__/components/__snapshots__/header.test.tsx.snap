// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should create a Header in english 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="wrapper"
    >
      <h1>
        <a
          href="/"
        >
          Mortality in France
        </a>
      </h1>
      <ul
        class="menu"
      >
        <li
          class=""
        >
          <a
            href="/overview"
          >
            Overview
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/comparison"
          >
            Comparison
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/distribution"
          >
            Distribution
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header should create a Header in french 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="wrapper"
    >
      <h1>
        <a
          href="/"
        >
          Mortalité en France
        </a>
      </h1>
      <ul
        class="menu"
      >
        <li
          class=""
        >
          <a
            href="/overview"
          >
            Vue d'Ensemble
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/comparison"
          >
            Comparaison
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/distribution"
          >
            Répartition
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
